src/Shell/Task/TestTask.php

```diff
@@ -513,6 +515,10 @@ public function generateConstructor($type, $fullClassName)
             $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');";
             $construct = "new {$className}(\$this->io);";
         }
+        if ($type === 'task') {
+            $pre = "\$this->io = \$this->getMock('Cake\Console\ConsoleIo');\n";
+            $construct = "\$this->getMock('{$fullClassName}', ['in', 'out', 'err', '_stop'], [\$this->io]);";
```

Comment:
I used the ``*TaskTest`` example of the core to build this : all of them mock these methods when building the test subject class. See : 

- [UnloadTaskTest](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Shell/Task/UnloadTaskTest.php#L38)
- [AssetsTaskTest](https://github.com/cakephp/cakephp/blob/master/tests/TestCase/Shell/Task/AssetsTaskTest.php#L45-L48)
- and all other classes.

But I can of course remove them if they are indeed not necessary. It was just to provide consistency with what is done in the core.